sympy/logic/boolalg.py

```diff
@@ -624,8 +656,15 @@ def __new__(cls, *args, **kwargs):
         elif len(argset) == 1:
             return argset.pop()
         elif True in argset:
-            argset.remove(True)
-            return Not(Xor(*argset))
+            frame = stack()[1][0]
+            frame_info = getframeinfo(frame)
```

Comment:
By default, if True is in argset, Xor's constructor removes True and returns Not. I use functions stack() and getframeinfo() to identify the name of the function which calls Xor's contructors. If function's name in ['_to_anf', '_distribute', 'ANF'], Xor's constructor returns Xor with True in argset.